09-schema/01-concepts.md

```diff
@@ -812,25 +828,27 @@ TypeQLDefine query = TypeQL.define(
 ```
 </div>
 
-In the example above the `person` was already defined as a subtype of an entity as well as first three attributes.
+In the example above, `person` was already defined as a subtype of `entity` and owning the first three attributes.
 
-You can even reuse your original query/file (e.g. `schema.tql`) that was used to create the original schema. 
-By adding concepts to the original query and executing it again you will achieve the same result — any concepts that 
-were created by the original request will not be created again (doubled) but the newly added concepts will be added to 
-the schema. Thus, running the same query a second time will have no effect.
+We can even reuse the original query/file (e.g. `schema.tql`) that was used to create the original schema. 
+By adding concepts to the original query and executing it again we will achieve the same result — any concepts that 
+were created by the original request will not be created again (duplicated) but the newly added concepts will be added 
+to the schema. Thus, running the same query a second time will have no effect (i.e. it is idempotent).
 
 <div class="note">
 [Important]
-You can only add concepts this way. Do not try to modify existing concepts or delete by modifying query and launching 
-it again.
+Since this operation is idempotent we can add any amount of previously applied statements that are already in the schema 
+into a define request. But what we can only add types this way. We can't modify or delete existing types by modifying 
+a define query and launching it again.
 </div>
 
 ### Deleting from a schema with Undefine
+
 As the keyword suggests, we use the `undefine` keyword to remove the definition of a type or its association with other types from the schema.
 
 <div class="note">
 [Important]
-Don't forget to `commit` after executing an `undefine` statement. Otherwise, anything you have undefined is NOT committed to the original database that is running on the TypeDB server.
+Don't forget to `commit` after executing an `undefine` statement. Otherwise, anything we have undefined is NOT committed to the original database that is running on the TypeDB server.
```

Comment:
```suggestion
Don't forget to `commit` after executing an `undefine` query. Otherwise, the resulting state of that query will NOT be committed to the database.
```

Reply:
If I understand correctly, there can be (and usually are) multiple queries in a define/undefine tql file. All of them go with a single transaction, but different queries. So what would we call this "text of the tql file with define in it"?

Reply:
Also, are we committing the state of the query? I mean is that a correct term? I thought we more like commit changes made by the query.

Reply:
> If I understand correctly, there can be (and usually are) multiple queries in a define/undefine tql file. All of them go with a single transaction, but different queries. So what would we call this "text of the tql file with define in it"?

I believe the whole file, if it is a single define statement, is one query.



> Also, are we committing the state of the query? I mean is that a correct term? I thought we more like commit changes made by the query.

You're right, this is poorly worded. Perhaps 'any changes will NOT [...]'. No need to go into more detail.